pandas/tests/test_frame.py

```diff
@@ -12926,6 +12926,34 @@ def test_stack_unstack(self):
         assert_frame_equal(unstacked_cols.T, self.frame)
         assert_frame_equal(unstacked_cols_df['bar'].T, self.frame)
 
+    def test_unstack_fill(self):
+
+        # GH #9746: fill_value keyword argument for Series
```

Comment:
add a test where the ``fill_value`` changes the dtype (so I think our current impl will error, so need to handle that)